src/OrleansProviders/Storage/MemoryStorage.cs

```diff
@@ -108,7 +108,7 @@ public virtual Task Init(string name, IProviderRuntime providerRuntime, IProvide
             for (int i = 0; i < numStorageGrains; i++)
             {
                 int idx = i; // Capture variable to avoid modified closure error
-                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => MemoryStorageGrainFactory.GetGrain(idx));
+                storageGrains[idx] = new Lazy<IMemoryStorageGrain>(() => providerRuntime.GrainFactory.GetGrain<IMemoryStorageGrain>(idx));
```

Comment:
No, I'm not saying that. :-) In my view, somebody writing a provider is likely to be already comfortable enough with the programming model to get GrainFactory from IProviderRuntime. If that's all ProviderBase provides, then I think it would be a nuisance rather than help. But if there's enough common functionality across providers that can be factored out, then ProviderBase would have value.